pkg/cluster/internal/logs/logs.go

```diff
@@ -63,7 +64,10 @@ func untar(logger log.Logger, r io.Reader, dir string) (err error) {
 
 		switch {
 		case err == io.EOF:
-			return nil
+			// drain the reader, which may have trailing null bytes
+			// we don't want to leave the writer hanging
+			_, err := io.Copy(ioutil.Discard, r)
```

Comment:
(tar writes in fixed size blocks, there is padding)

Reply:
excellent